Source/WebCore/rendering/RenderIterator.h

```diff
@@ -210,6 +275,28 @@ inline T* next(U& current, const RenderObject* stayWithin)
 
 } // namespace WebCore::RenderTraversal
 
+namespace RenderPostOrderTraversal {
+
+template <typename T, typename U>
+inline T* firstWithin(U& current)
+{
+    auto* descendant = RenderObjectPostOrderTraversal::firstLeafChild(current);
+    while (descendant && !isRendererOfType<T>(*descendant))
+        descendant = RenderObjectPostOrderTraversal::next(*descendant, &current);
+    return static_cast<T*>(descendant);
```

Comment:
You're right. I've fixed it the way you suggested. I also noticed that when an object has children it's iterated as a last child which is wrong. I've added another check to fix it. Also, I've made `RenderObjectPostOrderTraversal`s inlines non-templated since it's not really necessary.